spmd/tensor/parallel/api.py

```diff
@@ -251,6 +254,7 @@ def _parallelize_linear(
             _rowwise_parallelize_linear_fn,
             input_fn=parallel_style._prepare_input,  # type: ignore[arg-type, misc] # pyre-ignore[6]
             output_fn=parallel_style._prepare_output,  # type: ignore[arg-type, misc] # pyre-ignore[6]
+            input_shard_dim=-1,
```

Comment:
I think this should goes into prepare_input of rowwise parallel style directly?

Reply:
Is there a way for function to store a state?

Reply:
You _could_ store an attribute on the function itself. 
```
>>> def f():
>>>     pass
>>> f.a = 1
>>> f.a
1
```
I am not sure if this is recommended, but there _is_ precedent for this in [old FSDP code](https://github.com/pytorch/pytorch/blob/77d7f2c65945438e0292b270998cea07c0d9d3d8/torch/distributed/fsdp/wrap.py#L226-L227):
```
size_based_auto_wrap_policy.EXCLUDE_WRAP_MODULES = {nn.ModuleList, nn.ModuleDict}  # type: ignore[attr-defined]
size_based_auto_wrap_policy.FORCE_LEAF_MODULES = {nn.MultiheadAttention}  # type: ignore[attr-defined]
```
where `size_based_auto_wrap_policy` is a function.

Reply:
We will create a new prepare_input func instead.